### PR TITLE
Add assetGate

### PR DESCRIPTION
**Pull request by MarkSackerberg**
depends on https://github.com/metaplex-foundation/mpl-core-candy-machine/pull/14

### CODE REVIEW

**Review by tonyboylehub — APPROVED** [review]

**Comment by MarkSackerberg** [comment]
Will merge this when the CM deployment was done. Thank you @tonyboylehub

File: src/pages/core-candy-machine/guards/vanity-mint.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+API References: [create](https://mpl-core-candy-machine.typedoc.metaplex.com/functions/create.html), [VanityMint](https://mpl-core-candy-machine.typedoc.metaplex.com/types/VanityMint.html)
+
+{% /totem %}
+{% /dialect %}
+{% /dialect-switcher %}
+
+## Mint Settings
+
+_The Vanity Mint guard does not require mint settings. It expexts the mint address to match._
```
Comment (blockiosaurus) [inline src/pages/core-candy-machine/guards/vanity-mint.md:112]: Expexts typo

File: src/pages/core-candy-machine/guards/vanity-mint.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+{% /diagram %}
+
+## Guard Settings
+
+The Vanity Mint guard contains the following settings:
+
+- **Regular Expression**: A Regex that the mint address has to match. E.g. if you want all mints to start with string `mplx` you could use this as `regex` Parameter.
+
+{% dialect-switcher title="Set up a Candy Machine using the Asset Gate Guard" %}
```
Comment (blockiosaurus) [inline src/pages/core-candy-machine/guards/vanity-mint.md:89]: Copy pasta for asset gate guard

File: src/pages/core-candy-machine/guards/vanity-mint.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+{% dialect title="JavaScript" id="js" %}
+{% totem %}
+
+```ts
+create(umi, {
```
Comment (blockiosaurus) [inline src/pages/core-candy-machine/guards/vanity-mint.md:94]: Could you add the cases used in the test? 99% of users will probably only care about "starts with", "ends with", and "matches" so we should make copy-pastable examples if it's not too much extra work.

**Review by blockiosaurus — CHANGES_REQUESTED** [review]
Asset gate is perfect! Just some comments on Vanity mint

**Comment by MarkSackerberg** [comment]
@blockiosaurus thanks for the review!

I've added the examples in a list. (and added in addition "starts and ends with")

**Review by blockiosaurus — APPROVED** [review]
Perfect!

**Pull request merged**